xblock/core.py

```diff
@@ -90,6 +99,9 @@ def open_local_resource(cls, uri):
         unauthorized resource.
         """
 
+        if isinstance(uri, six.binary_type):
+            uri = uri.decode('utf-8')
```

Comment:
I'd rather have our methods be specific about what type of string they are going to take, rather than allowing both types.

Per https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax (because wikipedia is the authoritative source of everything), only ASCII characters are allowed.